narwhal/primary/src/core.rs

```diff
@@ -809,7 +652,54 @@ impl Core {
                 },
 
                 // We also receive here our new headers created by the `Proposer`.
-                Some(header) = self.rx_headers.recv() => self.process_own_header(header).await,
+                Some(header) = self.rx_headers.recv() => {
+                    let (tx_cancel, rx_cancel) = oneshot::channel();
+                    if let Some(cancel) = self.cancel_proposed_header {
+                        let _ = cancel.send(());
```

Comment:
The timing of this cancel is interesting, when we receive reconfiguration shutdown, we'll only check to cancel the last header after we receive another header from proposer? What is the desired behavior for pending headers during epoch change? Would we want to cancel them sooner?

Reply:
Reconfiguration shutdown cancels the last proposed header task and exits the entire core loop, not sure what you mean about it only happening after receiving another header?

As for epoch change, that works just like any round change. We will keep trying to complete a cert for the most recently proposed header until we (1) succeed, (2) fail permanently due to vote refusal, or (3) get a different one to try from the proposer.